### PR TITLE
Add makepython tutorials

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tutorial_code/EXAMPLES_LICENSE [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Zero-Clause BSD License
```
Comment (kasperl) [inline tutorial_code/EXAMPLES_LICENSE:1]: Maybe this should just be LICENSE given that nothing else is here?

File: tutorial_code/http/http.toit [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import http
```
Comment (kasperl) [inline tutorial_code/http/http.toit:1]: No copyright or license?

File: docs/tutorials/hardware/adc.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+(thus 4096 values) and ~0.14V is thus not just an imprecision.
+
+It turns out that the ADC is non-linear and can't really measure
+very low and high values:
+https://docs.espressif.com/projects/esp-idf/en/v4.2/esp32/api-reference/peripherals/adc.html#adc-calibration
+
+Keep this in mind when using the ADC.
+
+## Exercises
+
```
Comment (kasperl) [inline docs/tutorials/hardware/adc.mdx:108]: Drop newline for consistency.

File: docs/tutorials/hardware/binary.mdx [hunk 4]
```diff
@@ -0,0 +1,119 @@
+import BinaryDiagram from "../../_images/hw_ff_binary.png";
+import BinarySchematics from "../../_images/hw_ff_binary_schem.png";
+import Vibration from "../../_images/hw_vibration.jpg";
+import Pir from "../../_images/hw_pir.jpg";
+
+# Binary Sensors
```
Comment (kasperl) [inline docs/tutorials/hardware/binary.mdx:6]:
```suggestion
# Binary sensors
```

File: docs/tutorials/hardware/binary.mdx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+The sensor (SW-420) uses a spring inside the cylinder. So make sure to give
+it time to stabilize before attempting a new calibration.
+
+<img
+  src={Vibration}
+  alt="Vibration sensor"
+/>
+
+### IR Distance sensor
```
Comment (kasperl) [inline docs/tutorials/hardware/binary.mdx:95]:
```suggestion
### IR distance sensor
```

File: docs/tutorials/hardware/bme280.mdx [hunk 6]
```diff
@@ -0,0 +1,95 @@
+import Bme280Diagram from "../../_images/hw_ff_bme280.png";
+import Bme280Schematics from "../../_images/hw_ff_bme280_schem.png";
+
+# BME280 - Temperature, Humidity and Pressure Sensor
```
Comment (kasperl) [inline docs/tutorials/hardware/bme280.mdx:4]:
```suggestion
# BME280 - temperature, humidity and pressure sensor
```

File: docs/tutorials/hardware/button.mdx [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+If you have a button with 3 pins, typically with an integrated pull-down
+resistor, then you can just use two pins ("VCC" and "OUT").
+
+<img
+  src={HwButton}
+  alt="Photo of a button with 3 connectors"
+/>
+
+## Setup
+
```
Comment (kasperl) [inline docs/tutorials/hardware/button.mdx:32]: Drop newline for consistency.

File: docs/tutorials/hardware/dht11.mdx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+## Prerequisites
+We assume that you have set up your development environment as described
+in [the IDE tutorial](../../setup/ide).
+
+We also assume that you have flashed your device with Jaguar and that
+you are familiar with running Toit programs on it.
+If not, have a look at the [Hello world](../../setup/firstprogram) tutorial.
+
+## Setup
+
```
Comment (kasperl) [inline docs/tutorials/hardware/dht11.mdx:16]: Drop newline for consistency.

File: docs/tutorials/hardware/dht11.mdx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+connecting it. We have seen multiple DHT11 sensors being destroyed
+because they were connected the wrong way.
+
+<img
+  src={Dht11Diagram}
+  alt="DHT11 wiring diagram"
+/>
+
+## Packages
+
```
Comment (kasperl) [inline docs/tutorials/hardware/dht11.mdx:33]: Drop newline for consistency.

File: docs/tutorials/hardware/ds18b20.mdx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+## Prerequisites
+We assume that you have set up your development environment as described
+in [the IDE tutorial](../../setup/ide).
+
+We also assume that you have flashed your device with Jaguar and that
+you are familiar with running Toit programs on it.
+If not, have a look at the [Hello world](../../setup/firstprogram) tutorial.
+
+## Setup
+
```
Comment (kasperl) [inline docs/tutorials/hardware/ds18b20.mdx:27]: Drop newline for consistency.

File: docs/tutorials/hardware/ds18b20.mdx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  alt="Ds18b20 wiring diagram"
+/>
+
+<img
+  src={Ds18b20Schematics}
+  alt="Ds18b20 wiring schematics"
+/>
+
+## Packages
+
```
Comment (kasperl) [inline docs/tutorials/hardware/ds18b20.mdx:48]: Drop newline for consistency.

File: docs/tutorials/hardware/ds18b20.mdx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+main:
+  pin := gpio.Pin GPIO_PIN_NUM
+  ow := one_wire.Protocol pin
+  driver := ds18b20.Driver ow
+
+  (Duration --ms=500).periodic:
+    print "Temperature: $(%.2f driver.read_temperature) C"
+```
+
+Exercises
```
Comment (kasperl) [inline docs/tutorials/hardware/ds18b20.mdx:96]:
```suggestion
## Exercises
```

File: docs/tutorials/hardware/i2c.mdx [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+We also assume that you have flashed your device with Jaguar and that
+you are familiar with running Toit programs on it.
+If not, have a look at the [Hello world](../../setup/firstprogram) tutorial.
+
+While not necessary, we recommend to have a look at the [LED](../led)
+tutorial to get familiar with the basics of using the GPIO pins.
+
+## Introduction
+
```
Comment (kasperl) [inline docs/tutorials/hardware/i2c.mdx:16]: Drop newline for consistency.

File: docs/tutorials/hardware/i2c.mdx [hunk 14]
```diff
@@ -0,0 +1,82 @@
+# I2C Bus
```
Comment (kasperl) [inline docs/tutorials/hardware/i2c.mdx:1]:
```suggestion
# I2C bus
```

File: docs/tutorials/hardware/led.mdx [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+## Prerequisites
+We assume that you have set up your development environment as described
+in [the IDE tutorial](../../setup/ide).
+
+We also assume that you have flashed your device with Jaguar and that
+you are familiar with running Toit programs on it.
+If not, have a look at the [Hello world](../../setup/firstprogram) tutorial.
+
+## Setup
+
```
Comment (kasperl) [inline docs/tutorials/hardware/led.mdx:16]: Drop newline for consistency.

File: docs/tutorials/hardware/neopixel.mdx [hunk 16]
```diff
@@ -0,0 +1,149 @@
+import NeopixelDiagram from "../../_images/hw_ff_neopixel.png";
+import NeopixelSchematics from "../../_images/hw_ff_neopixel_schem.png";
+
+# Neopixel
+
```
Comment (kasperl) [inline docs/tutorials/hardware/neopixel.mdx:5]: Drop newline for consistency.

File: docs/tutorials/hardware/neopixel.mdx [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+## Prerequisites
+We assume that you have set up your development environment as described
+in [the IDE tutorial](../../setup/ide).
+
+We also assume that you have flashed your device with Jaguar and that
+you are familiar with running Toit programs on it.
+If not, have a look at the [Hello world](../../setup/firstprogram) tutorial.
+
+## Setup
+
```
Comment (kasperl) [inline docs/tutorials/hardware/neopixel.mdx:30]: Drop newline for consistency.

File: docs/tutorials/hardware/neopixel.mdx [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+  alt="Wiring diagram for a Neopixel LED strip"
+/>
+
+<img
+  src={NeopixelSchematics}
+  alt="Schematics for a Neopixel LED strip"
+/>
+
+## Packages
+
```
Comment (kasperl) [inline docs/tutorials/hardware/neopixel.mdx:49]: Drop newline for consistency.

File: docs/tutorials/hardware/neopixel.mdx [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+<Note>
+
+You can probably just write `jag pkg install pixel_strip`, but the full ID together
+with the version is more explicit, and will make sure you get the right package.
+
+</Note>
+
+## Code
+
```
Comment (kasperl) [inline docs/tutorials/hardware/neopixel.mdx:69]: Drop newline for consistency.

File: docs/tutorials/hardware/neopixel.mdx [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+dim all lights.
+
+Then we set the current LED to full brightness.
+
+Finally, we send the new values to the strip, and wait 70ms.
+
+Before the next iteration we move to the next LED.
+
+## Exercises
+
```
Comment (kasperl) [inline docs/tutorials/hardware/neopixel.mdx:134]: Drop newline for consistency.

File: docs/tutorials/hardware/pwm_buzzer.mdx [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+  alt="Buzzer wiring diagram"
+/>
+
+<img
+  src={BuzzerSchematics}
+  alt="Buzzer schematics"
+/>
+
+## Code
+
```
Comment (kasperl) [inline docs/tutorials/hardware/pwm_buzzer.mdx:32]: Drop newline for consistency.

File: docs/tutorials/hardware/pwm_led.mdx [hunk 22]
```diff
@@ -0,0 +1,159 @@
+import LedDiagram from "../../_images/hw_ff_led.png";
+import LedSchematics from "../../_images/hw_ff_led_schem.png";
+import PwmWiki from "../../_images/pwm.png";
+
+# Fading Light
+
```
Comment (kasperl) [inline docs/tutorials/hardware/pwm_led.mdx:6]: Drop newline for consistency.

File: docs/tutorials/hardware/pwm_led.mdx [hunk 23]
```diff
@@ -0,0 +1,159 @@
+import LedDiagram from "../../_images/hw_ff_led.png";
+import LedSchematics from "../../_images/hw_ff_led_schem.png";
+import PwmWiki from "../../_images/pwm.png";
+
+# Fading Light
```
Comment (kasperl) [inline docs/tutorials/hardware/pwm_led.mdx:5]:
```suggestion
# Fading light
```

File: docs/tutorials/hardware/pwm_servo.mdx [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+  alt="Servo motor diagram"
+/>
+
+<img
+  src={ServoSchematics}
+  alt="Servo motor schematics"
+/>
+
+## Code
+
```
Comment (kasperl) [inline docs/tutorials/hardware/pwm_servo.mdx:33]: Drop newline for consistency.

File: docs/tutorials/hardware/ssd1306.mdx [hunk 25]
```diff
@@ -0,0 +1,269 @@
+import Ssd1306Diagram from "../../_images/hw_ff_ssd1306.png";
+import Ssd1306Schematics from "../../_images/hw_ff_ssd1306_schem.png";
+import Makerfabs from "../../_images/hw_makerfabs.jpg";
+
+# OLED Display
```
Comment (kasperl) [inline docs/tutorials/hardware/ssd1306.mdx:5]:
```suggestion
# OLED display
```

File: docs/tutorials/hardware/ssd1306.mdx [hunk 26]
```diff
@@ -0,0 +1,269 @@
...
+In this tutorial we will write text on it, and draw some icons.
+
+## Prerequisites
+This tutorial assumes that you have done the [i2c](../i2c) tutorial.
+
+We recommend to have read the [packages](../../setup/packages) tutorial,
+since we are going to use a package for the SSD1306 driver.
+
+## Setup
+
```
Comment (kasperl) [inline docs/tutorials/hardware/ssd1306.mdx:18]: Drop newline for consistency.

File: docs/tutorials/hardware/ultra.mdx [hunk 27]
```diff
@@ -0,0 +1,158 @@
...
+
+We also assume that you have flashed your device with Jaguar and that
+you are familiar with running Toit programs on it.
+If not, have a look at the [Hello world](../../setup/firstprogram) tutorial.
+
+We recommend that you have finished the [LED tutorial](../led) before
+starting this tutorial.
+
+## Setup
+
```
Comment (kasperl) [inline docs/tutorials/hardware/ultra.mdx:22]: Drop newline for consistency.

File: docs/tutorials/index.mdx [hunk 28]
```diff
@@ -1,7 +1,244 @@
...
+Use binary sensors.
+</Box>
+
+<Box title="Analog to Digital Converter" to="hardware/adc">
+<img
+  src={Adc}
+  alt="The symbol for an Analog to Digital Converter."
+/>
+
+Use an Analog to Digital Converter (ADC).
```
Comment (kasperl) [inline docs/tutorials/index.mdx:141]:
```suggestion
Use an analog-to-digital Converter (ADC).
```

File: docs/tutorials/setup/firstprogram.mdx [hunk 29]
```diff
@@ -0,0 +1,91 @@
+import HelloJagRun from "../../_images/hello_jag_run.png";
+import HelloJagWatch from "../../_images/hello_jag_watch.png";
+
+# Hello world
+
```
Comment (kasperl) [inline docs/tutorials/setup/firstprogram.mdx:5]: Drop newline for consistency.

File: docs/tutorials/setup/ide.mdx [hunk 30]
```diff
@@ -0,0 +1,190 @@
...
+import VscodeToitExtension from "../../_images/setup_vscode_toit_extension.png";
+import JagPort from "../../_images/setup_jag_port.png";
+import JagWiFi from "../../_images/setup_jag_wifi.png";
+import JagFlash from "../../_images/setup_jag_flash.png";
+import JagMonitor from "../../_images/setup_jag_monitor.png";
+import JagScan from "../../_images/setup_jag_scan.png";
+import VscodeSplitTerminals from "../../_images/setup_split_terminals.png";
+
+# IDE Setup
+
```
Comment (kasperl) [inline docs/tutorials/setup/ide.mdx:12]: Drop newline for consistency.

File: docs/tutorials/setup/ide.mdx [hunk 31]
```diff
@@ -0,0 +1,190 @@
...
+import VscodeExtension from "../../_images/setup_vscode_extension.png";
+import VscodeToitExtension from "../../_images/setup_vscode_toit_extension.png";
+import JagPort from "../../_images/setup_jag_port.png";
+import JagWiFi from "../../_images/setup_jag_wifi.png";
+import JagFlash from "../../_images/setup_jag_flash.png";
+import JagMonitor from "../../_images/setup_jag_monitor.png";
+import JagScan from "../../_images/setup_jag_scan.png";
+import VscodeSplitTerminals from "../../_images/setup_split_terminals.png";
+
+# IDE Setup
```
Comment (kasperl) [inline docs/tutorials/setup/ide.mdx:11]:
```suggestion
# Setting up the IDE
```

or

```suggestion
# IDE setup
```

File: docs/tutorials/setup/ide.mdx [hunk 32]
```diff
@@ -0,0 +1,190 @@
...
+Search for Toit and install the extension.
+
+<img
+  src={VscodeToitExtension}
+  alt="Screenshot of the Toit extension page"
+/>
+
+
+## Open a folder
+
```
Comment (kasperl) [inline docs/tutorials/setup/ide.mdx:92]: Drop newline for consistency.

File: docs/tutorials/setup/ide.mdx [hunk 33]
```diff
@@ -0,0 +1,190 @@
...
+``` bash
+jag scan 192.168.86.25
+```
+
+Your development environment is now ready to use. You can start writing
+programs in Toit and run them on your device.
+
+## Next steps
+
```
Comment (kasperl) [inline docs/tutorials/setup/ide.mdx:188]: Drop newline for consistency.

File: docs/tutorials/setup/packages.mdx [hunk 34]
```diff
@@ -0,0 +1,136 @@
+# Packages
+
```
Comment (kasperl) [inline docs/tutorials/setup/packages.mdx:2]: Drop newline for consistency.

**Review by kasperl — APPROVED** [review]
Looks good to me!

File: docs/tutorials/hardware/pwm_buzzer.mdx [hunk 35]
```diff
@@ -0,0 +1,76 @@
+import BuzzerDiagram from "../../_images/hw_ff_buzzer.png";
```
Comment (kasperl) [inline docs/tutorials/hardware/pwm_buzzer.mdx:1]: Not sure we typically use underscores (_) in the mdx file names? I think we typically stick to dashes (-).

This applies to 3 pwm files, I think.

File: docs/tutorials/hardware/adc.mdx (hunk 3 above)
Comment (floitsch) [reply docs/tutorials/hardware/adc.mdx]: done.

File: docs/tutorials/hardware/binary.mdx (hunk 4 above)
Comment (floitsch) [reply docs/tutorials/hardware/binary.mdx]: done.

File: docs/tutorials/hardware/binary.mdx (hunk 5 above)
Comment (floitsch) [reply docs/tutorials/hardware/binary.mdx]: done.

File: docs/tutorials/hardware/bme280.mdx (hunk 6 above)
Comment (floitsch) [reply docs/tutorials/hardware/bme280.mdx]: done.

File: docs/tutorials/hardware/button.mdx (hunk 7 above)
Comment (floitsch) [reply docs/tutorials/hardware/button.mdx]: done.

File: docs/tutorials/hardware/dht11.mdx (hunk 8 above)
Comment (floitsch) [reply docs/tutorials/hardware/dht11.mdx]: done.

File: docs/tutorials/hardware/dht11.mdx (hunk 9 above)
Comment (floitsch) [reply docs/tutorials/hardware/dht11.mdx]: done.

File: docs/tutorials/hardware/ds18b20.mdx (hunk 10 above)
Comment (floitsch) [reply docs/tutorials/hardware/ds18b20.mdx]: done.

File: docs/tutorials/hardware/ds18b20.mdx (hunk 11 above)
Comment (floitsch) [reply docs/tutorials/hardware/ds18b20.mdx]: done.

File: docs/tutorials/hardware/ds18b20.mdx (hunk 12 above)
Comment (floitsch) [reply docs/tutorials/hardware/ds18b20.mdx]: done.

File: docs/tutorials/hardware/i2c.mdx (hunk 14 above)
Comment (floitsch) [reply docs/tutorials/hardware/i2c.mdx]: done.

File: docs/tutorials/hardware/i2c.mdx (hunk 13 above)
Comment (floitsch) [reply docs/tutorials/hardware/i2c.mdx]: done.

File: docs/tutorials/hardware/led.mdx (hunk 15 above)
Comment (floitsch) [reply docs/tutorials/hardware/led.mdx]: done.

File: docs/tutorials/hardware/neopixel.mdx (hunk 16 above)
Comment (floitsch) [reply docs/tutorials/hardware/neopixel.mdx]: done.

File: docs/tutorials/hardware/neopixel.mdx (hunk 17 above)
Comment (floitsch) [reply docs/tutorials/hardware/neopixel.mdx]: done.

File: docs/tutorials/hardware/neopixel.mdx (hunk 18 above)
Comment (floitsch) [reply docs/tutorials/hardware/neopixel.mdx]: done.

File: docs/tutorials/hardware/neopixel.mdx (hunk 19 above)
Comment (floitsch) [reply docs/tutorials/hardware/neopixel.mdx]: done.

File: docs/tutorials/hardware/neopixel.mdx (hunk 20 above)
Comment (floitsch) [reply docs/tutorials/hardware/neopixel.mdx]: done.

File: docs/tutorials/hardware/pwm_buzzer.mdx (hunk 35 above)
Comment (floitsch) [reply docs/tutorials/hardware/pwm_buzzer.mdx]: done.

File: docs/tutorials/hardware/pwm_buzzer.mdx (hunk 21 above)
Comment (floitsch) [reply docs/tutorials/hardware/pwm_buzzer.mdx]: done.

File: docs/tutorials/hardware/pwm_led.mdx (hunk 23 above)
Comment (floitsch) [reply docs/tutorials/hardware/pwm_led.mdx]: done.

File: docs/tutorials/hardware/pwm_led.mdx (hunk 22 above)
Comment (floitsch) [reply docs/tutorials/hardware/pwm_led.mdx]: done.

File: docs/tutorials/hardware/pwm_servo.mdx (hunk 24 above)
Comment (floitsch) [reply docs/tutorials/hardware/pwm_servo.mdx]: done.

File: docs/tutorials/hardware/ssd1306.mdx (hunk 25 above)
Comment (floitsch) [reply docs/tutorials/hardware/ssd1306.mdx]: done.

File: docs/tutorials/hardware/ssd1306.mdx (hunk 26 above)
Comment (floitsch) [reply docs/tutorials/hardware/ssd1306.mdx]: done.

File: docs/tutorials/hardware/ultra.mdx (hunk 27 above)
Comment (floitsch) [reply docs/tutorials/hardware/ultra.mdx]: done.

File: docs/tutorials/index.mdx (hunk 28 above)
Comment (floitsch) [reply docs/tutorials/index.mdx]: done.

File: docs/tutorials/setup/firstprogram.mdx (hunk 29 above)
Comment (floitsch) [reply docs/tutorials/setup/firstprogram.mdx]: done.

File: docs/tutorials/setup/ide.mdx (hunk 31 above)
Comment (floitsch) [reply docs/tutorials/setup/ide.mdx]: done.

File: docs/tutorials/setup/ide.mdx (hunk 30 above)
Comment (floitsch) [reply docs/tutorials/setup/ide.mdx]: done.

File: docs/tutorials/setup/ide.mdx (hunk 32 above)
Comment (floitsch) [reply docs/tutorials/setup/ide.mdx]: done.

File: docs/tutorials/setup/ide.mdx (hunk 33 above)
Comment (floitsch) [reply docs/tutorials/setup/ide.mdx]: done.

File: docs/tutorials/setup/packages.mdx (hunk 34 above)
Comment (floitsch) [reply docs/tutorials/setup/packages.mdx]: done.

File: tutorial_code/EXAMPLES_LICENSE (hunk 1 above)
Comment (floitsch) [reply tutorial_code/EXAMPLES_LICENSE]: Renamed to LICENSE_BSD0 (similar to what we have in the docs folder).

File: tutorial_code/http/http.toit (hunk 2 above)
Comment (floitsch) [reply tutorial_code/http/http.toit:5]: done.

**Pull request merged**